hack/ci-e2e.sh

```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
```

Comment:
We do not have a license/copyright notice on almost any file in this repo. If that is required we should first add it to all files and then add a test to make sure that any new files include it.
The only reason I put the copyright notice in `ensure_kubectl.sh` is because I copied it from CAPM3 and it would be wrong to remove it.

Reply:
If repo has top-level LICENCE file, it applies to all files not having explicit copyright notice or some other directory level LICENCE file overriding it. It is not mandatory to have it. If we want to have it in each file, then we must do what Lennart suggested, have a test for it to enforce.

Reply:
> I found that theres a boilerplate.go.txt in the hack folders but I am not sure how that should work and it also seems to be missing the copyright line.

We have a test for this in CAPO. This is how it looks: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/hack/verify-boilerplate.sh
I would suggest we do not add any more in this PR and instead handle all the files in a separate PR where we also add a check for it. Otherwise we will easily end up with small differences that are hard to spot. In the worst case, someone puts a completely different copy right notice on a file without us noticing. From that perspective it is much "safer" to just have the repo license until we have an automated check for this.

Reply:
I can see a similar file in CAPM3 as well, I will check how it works https://github.com/metal3-io/cluster-api-provider-metal3/blob/main/hack/verify-boilerplate.sh